docs_src/microservices/device/profile/Ch-DeviceProfileRef.md

```diff
@@ -0,0 +1,117 @@
+# Device Profile Reference
+
+This chapter details the structure of a Device Profile and allowable values for
+its fields.
+
+Device Profile
+--------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+Name | String | Y | Must be unique in the EdgeX deployment
+Manufacturer | String | N |
+Model | String | N |
+Labels | Array of String | N |
+DeviceResources | Array of DeviceResource | Y |
+DeviceCommands  | Array of DeviceCommand | N |
+CoreCommands | Array of CoreCommand | N |
+
+DeviceResource
+--------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+Name | String | Y | Must be unique in the EdgeX deployment
+Tag | String | N
+Attributes | String-String Map | Y | Each Device Service should define required and optional keys
+Properties | ProfileProperty | Y |
+
+ProfileProperty
+---------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+Value | PropertyValue | Y |
+Units | Units | N |
+
+PropertyValue
+-------------
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+Type | Enum | Y | `uint8`, `uint16`, `uint32`, `uint64`, `int8`, `int16`, `int32`, `int64`, `float32`, `float64`, `bool`, `string`, `binary`, `uint8array`, `uint16array`, `uint32array`, `uint64array`, `int8array`, `int16array`, `int32array`, `int64array`, `float32array`, `float64array`, `boolarray`
+ReadWrite | Enum | Y | `R`, `W`, `RW`
+DefaultValue | String | N | If present, should be compatible with the Type field
+Mask | Unsigned Int | N | Only valid where Type is one of the integer types
+Shift | Unsigned Int | N | Only valid where Type is one of the integer types
+Scale | Int or Float | N | Only valid where Type is one of the integer or float types
+Offset | Int or Float | N | Only valid where Type is one of the integer or float types
+Base | Int or Float | N | Only valid where Type is one of the integer or float types
+Assertion | String | N |
+FloatEncoding | Enum | N | `base64`, `eNotation` - Only valid where Type is one of the float types
+MediaType | String | N | Only valid where Type is Binary
+
+Units
+-----
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+DefaultValue | String | Y |
+
+DeviceCommand
+-------------
+
+(NB represented in Go by `ProfileResource`)
+
+Field Name | Type | Required? | Notes
+:--- | :--- | :--- | :---
+Name | String | Y | Must be unique in this profile. May have the same name as a DeviceResource but this will make the DeviceResource not accessible individually
```

Comment:
This is odd behavior which will trip up some users.  We need to either explain why one would not want a DeviceResource to be individually accessible or disallow name collisions between ProfileResource and DeviceResource.

Reply:
Noted that using the same name is only really useful if you want to apply Mappings on the resource.